src/eth_provider/provider.rs

```diff
@@ -138,11 +139,105 @@ pub struct EthDataProvider<SP: starknet::providers::Provider> {
     chain_id: u64,
 }
 
-impl<SP: starknet::providers::Provider> EthDataProvider<SP> {
+impl<SP> EthDataProvider<SP>
+where
+    SP: starknet::providers::Provider + Send + Sync,
+{
     /// Returns a reference to the database.
     pub fn database(&self) -> &Database {
         &self.database
     }
+
+    pub async fn retry_transactions(&self) -> EthProviderResult<Vec<B256>> {
```

Comment:
can we maybe move this towards the end of the file in the other `impl<SP> EthDataProvider<SP>`?

Reply:
Yes just fixed that